src/Type/MixedType.php

```diff
@@ -472,8 +474,8 @@ public function toBoolean(): BooleanType
 	public function toNumber(): Type
 	{
 		return new UnionType([
-			$this->toInteger(),
-			$this->toFloat(),
+			new IntegerType(),
+			new FloatType(),
```

Comment:
required change to prevent

```
     Internal error: Cannot create PHPStan\Type\UnionType with: int<min, -1>|int<1, max>, float while analysing file                                                            
     /Users/staabm/workspace/phpstan-src/src/PhpDoc/TypeNodeResolver.php                                                                                                        
     Post the following stack trace to https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml:                                                                  
     ## /Users/staabm/workspace/phpstan-src/src/Type/UnionType.php(60)                                                                                                          
     #0 /Users/staabm/workspace/phpstan-src/src/Type/UnionType.php(77): PHPStan\Type\UnionType::PHPStan\Type\{closure}()                                                        
     #1 /Users/staabm/workspace/phpstan-src/src/Type/MixedType.php(481): PHPStan\Type\UnionType->__construct(Array)                                                             
     #2 /Users/staabm/workspace/phpstan-src/src/Reflection/InitializerExprTypeResolver.php(698): PHPStan\Type\MixedType->toNumber()                                             
     #3 /Users/staabm/workspace/phpstan-src/src/Analyser/MutatingScope.php(1041):        
     ```